File: backend/app/schemas/platform.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class PlatformBase(BaseModel):
    code: str
    name: str
    platform_type: Optional[str] = None


class PlatformCreate(PlatformBase):
    pass


class PlatformUpdate(BaseModel):
    name: Optional[str] = None
    platform_type: Optional[str] = None


class PlatformResponse(PlatformBase):
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/platform.ts
export interface Platform {
  code: string;
  name: string;
  platform_type?: string;
  created_at?: string;
}

export interface PlatformCreate {
  code: string;
  name: string;
  platform_type?: string;
}

export interface PlatformUpdate {
  name?: string;
  platform_type?: string;
}
